SPEC.md

```diff
@@ -130,6 +134,11 @@ The network configuration is described in JSON form. The configuration can be st
   - `routes` (list): List of subnets (in CIDR notation) that the CNI plugin should ensure are reachable by routing them through the network. Each entry is a dictionary containing:
     - `dst` (string): subnet in CIDR notation
     - `gw` (string): IP address of the gateway to use. If not specified, the default gateway for the subnet is assumed (as determined by the IPAM plugin).
+- `dns`: Dictionary with DNS specific values:
+  - `nameservers` (list of strings): list of a priority-ordered list of DNS nameservers that this network is aware of. Each entry in the list is a string containing either an IPv4 or an IPv6 address.
+  - `domain` (string): the local domain used for short hostname lookups.
```

Comment:
Definitely; in most (all?) cases the underlying DNS implementation will enforce some kind of mutual exclusivity, so that should be reflected in the docs here.  Plugins cannot expect that giving both will actually use both, unless there's some hackery elsewhere.  So it should be defined explicitly what the precedence is.

I vote for saying something like "if search is given, it is preferred over domain".